backend/src/TodosDatabase.ts

```diff
@@ -0,0 +1,64 @@
+import { TodoDto } from './dto/todoDto';
+import { ConnectionService } from './app.connectionService';
+
+export default class TodosDatabase {
+  private static connectionService = new ConnectionService();
+
+  static async createTodo(todo: TodoDto) {
+    const newTodoOrder = await this.getNextTodoMaxOrder();
+    console.log(newTodoOrder);
+    const insertTodoQuery = `
+    INSERT INTO "todosTable" (id,title,date,status,todosorder)
+    VALUES ($1,$2,$3,$4,$5)
+    RETURNING *;`;
+    const todoValues = [
+      todo.id,
+      todo.title,
+      todo.date,
+      todo.status,
+      newTodoOrder,
+    ];
+    return this.connectionService.query(insertTodoQuery, todoValues);
+  }
+
+  private static async getNextTodoMaxOrder(): Promise<number> {
+    const maxOrderQuery = `SELECT * FROM "todosTable" ORDER BY todosorder DESC LIMIT 1`;
+    const resultRows = (await this.connectionService.query(maxOrderQuery)).rows;
+    const maxOrder = !resultRows.length
+      ? 0
+      : (resultRows[0]['todosorder'] as number);
+    return maxOrder + 1;
+  }
+
+  static async getAllTodos() {
+    const selectAllQuery = `SELECT * FROM "todosTable" ORDER BY todosorder ASC;`;
+    return this.connectionService.query(selectAllQuery);
+  }
+
+  static async getTodoById(id: string) {
+    const selectTodoQuery = `SELECT * FROM "todosTable" WHERE id='${id}';`;
+    return this.connectionService.query(selectTodoQuery);
+  }
+
+  static async updateTodo(updatedTodo: TodoDto) {
+    const todoAttributesPairs = Object.entries(updatedTodo)
+      .filter(([key, value]) => key !== 'id')
+      .map(([key, value]) => `${key}='${value}'`)
+      .join(', ');
+    const updateTodoQuery = `UPDATE "todosTable" SET ${todoAttributesPairs} WHERE id='${updatedTodo.id}' RETURNING *;`;
+    return this.connectionService.query(updateTodoQuery);
```

Comment:
i kinda like it but usually we prefer to be explicit in our queries, e.g. less sql code generation, write the query explicitly 
note that you can merge both update and add methods in one `upsert` method, look for examples in our codebase.